plugin/evm/limit_order.go

```diff
@@ -74,8 +74,10 @@ func NewLimitOrderProcesser(ctx *snow.Context, txPool *txpool.TxPool, shutdownCh
 	filterAPI := filters.NewFilterAPI(filterSystem)
 
 	// need to register the types for gob encoding because memory DB has an interface field(ContractOrder)
-	gob.Register(&hu.LimitOrder{})
-	gob.Register(&hu.IOCOrder{})
+	// naming hu.LimitOrder as orderbook.LimitOrder instead of hubbleutils.LimitOrder because of backward compatibility. DO NOT CHANGE
+	// same for hu.IOCOrder
+	gob.RegisterName("*orderbook.LimitOrder", &hu.LimitOrder{})
+	gob.RegisterName("*orderbook.IOCOrder", &hu.IOCOrder{})
```

Comment:
But I only changed this as a trial, snapshot didn't recover even when this was unchaged.

Reply:
in the previous version, the struct was registered as "*orderbook.LimitOrder" cuz it derived the name from the struct. The type `hubbleutils.LimitOrder`(new struct) is identical but has a different name. So we can provide the new struct with the old name. In the snapshots saved in previous version the struct would have the name "*orderbook.LimitOrder", so the new version also needs to have the same name registered.